### ISSUE TITLE
Vulkan HW decoder doesn't work on Flatpak due to MESA Radv driver bug

### ISSUE THREAD

**Issue by streetpea** [streetpea/chiaki4deck#359]
**Describe the bug**
A bunch of colors show up instead of the stream when using hw decoder or flatpak. This appears to be a bug with the June 10th update of org.freedesktop.Platform.GL.default/x86_64/23.08-extra

**To Reproduce**
Launch flatpak using vulkan hw decoder or auto hw decoder option with the flatpak and see lots of colors instead of a stream

**Expected behavior**
See the stream

**Alternatives**
Use vaapi hw decoder

**Resolution**
Either don’t use Vulkan as hw decoder (switch to auto with latest flatpak or vaapi) or Downgrade to previous version of GL ...
System version
```
sudo flatpak update --commit=2a1f67b928d981dcc265f90c943cbbecd6ec70ab9a75dcfd24d69b91b2848e1b org.freedesktop.Platform.GL.default/x86_64/23.08-extra
```
User version (if installed)
```
flatpak update --user  --commit=2a1f67b928d981dcc265f90c943cbbecd6ec70ab9a75dcfd24d69b91b2848e1b org.freedesktop.Platform.GL.default/x86_64/23.08-extra
```

also if you don't want to be prompted for the upgrade you can mask with

```
sudo flatpak mask org.freedesktop.Platform.GL.default/x86_64/23.08-extra
```

And for the user version
```
flatpak mask --user org.freedesktop.Platform.GL.default/x86_64/23.08-extra
```

*Note: You can always unmask at any time with flatpak unmask for example*
```
flatpak unmask --user org.freedesktop.Platform.GL.default/x86_64/23.08-extra
```

**Comment by Tonyb681** [comment]
I’m sorry, but I’m new to this and am having this same issue. How would I downgrade to the previous version of GL? Thank you

**Comment by streetpea** [comment]
@Tonyb681 run the command above starting with sudo flatpak in my comment in the konsole.

**Comment by Tonyb681** [comment]
Thank you for the response. I ran the command and did the downgrade, which promptly resulted in an update request (good sign), but I am still getting the same issue. I can hear everything, but the picture is messed up.
![IMG_2477](https://github.com/streetpea/chiaki4deck/assets/172431807/cbd40b43-0e7e-4981-af21-4a9f6750e336)

**Comment by streetpea** [comment]
@Tonyb681 it’s possible you have the user version installed and that would override the system version (so in essence the package you are actually using isn’t being downgraded). Please run the same command but with user so
```
flatpak update --user  --commit=2a1f67b928d981dcc265f90c943cbbecd6ec70ab9a75dcfd24d69b91b2848e1b org.freedesktop.Platform.GL.default/x86_64/23.08-extra
``

**Comment by Tonyb681** [comment]
Perfect. Thank you so much for the quick responses. This has fixed my issues.

**Comment by streetpea** [comment]
@Tonyb681 also if you don't want to be prompted for the upgrade you can mask with

```
sudo flatpak mask org.freedesktop.Platform.GL.default/x86_64/23.08-extra
```

And for the user version
```
flatpak mask --user org.freedesktop.Platform.GL.default/x86_64/23.08-extra
```

You can always unmask at any time with flatpak unmask for example
```

flatpak unmask --user
org.freedesktop.Platform.GL.default/x86_64/23.08-extra
```

**Comment by streetpea** [comment]
Auto updated to use vaapi on flatpak. Now if you set hw decoder to Auto and update your flatpaks you won’t be using Vulkan so won’t hit the bug.

**Comment by streetpea** [comment]
Pr fixing mesa bug: https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/29688

**Comment by streetpea** [comment]
Fix was released with mesa version 24.1.2 https://docs.mesa3d.org/relnotes/24.1.2.html. Now, we're just waiting for the platform.gl-default to pull in the newer mesa version

**Comment by streetpea** [comment]
Fix is now live in the latest flatpak version of `org.freedesktop.Platform.GL.default/x86_64/23.08-extra`
